mlkem/native/aarch64/ntt_clean.S

```diff
@@ -268,11 +292,6 @@ layer4567_start:
         ct_butterfly_v data0, data1, root1, root1_tw
         ct_butterfly_v data2, data3, root2, root2_tw
 
-        barrett_reduce data0
```

Comment:
Err.. why have these been removed? Why is that OK?

Reply:
That's what the PR is about!

Reply:
OK... so we're documenting the bounds _and_ removing redundant reductions as a side-effect of that.